lib/extensions/descendant_loader.rb

```diff
@@ -165,8 +165,14 @@ def cache_path
 
     def load_cache
       return unless cache_path.exist?
-      data = YAML.load_file(cache_path)
+
+      data = File.open(cache_path, "r") do |f|
+        f.flock(File::LOCK_SH)
```

Comment:
NOTE we need someone with OSX to test this out

Reply:
```ruby
DescendantLoader.instance.load_cache
# => nil
```

Do you need more than this?